### PR TITLE
Sharon unhandled special issues

### PR DESCRIPTION
**Pull request by sharonwarner**
This PR brings up an email page for special issues that we can't create an EP for. It fills in the email address of the RO the task is sent to. 

Connects #722 

- [ ] Confirmed that the routing office and email address is set correctly for PMC issues, education issues, and COWC issues. 
- [ ] Confirmed UI matches the mockups. 
screen shot 2017-02-20 at 9 13 28 pm

### CODE REVIEW

File: app/controllers/establish_claims_controller.rb [hunk 1]
```diff
@@ -20,6 +20,11 @@ def note_complete
     render json: {}
   end
 
+  def email_complete
+    task.complete!(status: 5)
```
Comment (joofsh) [inline app/controllers/establish_claims_controller.rb]: would it be possible to reference the `Task::COMPLETION_STATUS_MAPPING` here for clarity?

File: app/models/task.rb [hunk 2]
```diff
@@ -95,7 +96,7 @@ def completion_status_code(text)
     end
 
     event :complete do
-      transitions from: :reviewed, to: :completed, after: proc { |*args| save_completion_status(*args) }
+      transitions from: [:started, :reviewed], to: :completed, after: proc { |*args| save_completion_status(*args) }
```
Comment (joofsh) [inline app/models/task.rb]: is there a conditional guard we can add here for safety?  We only want to allow transitioning directly from started => completed in a couple specific scenarios, right?  I think you can add an `, if:  conditional_method`

File: client/app/containers/EstablishClaimPage/EstablishClaim.jsx [hunk 3]
```diff
@@ -362,6 +374,30 @@ export default class EstablishClaim extends BaseForm {
...
+    this.setState({
+      loading: true
+    });
+
+    return ApiUtil.post(`/dispatch/establish-claim/${task.id}/email-complete`).
+        then(() => {
+          this.reloadPage();
+        }, () => {
+          handleAlert(
+        'error',
```
Comment (joofsh) [inline client/app/containers/EstablishClaimPage/EstablishClaim.jsx]: minor nit: the indentation looks off here

File: client/app/containers/EstablishClaimPage/EstablishClaimEmail.jsx [hunk 4]
```diff
@@ -0,0 +1,106 @@
...
+  constructor(props) {
+    super(props);
+    let {
+      appeal,
+      specialIssues
+    } = props;
+
+    let selectedSpecialIssue = Object.keys(specialIssues).
+      filter((key) => specialIssues[key].value).
+      map((key) => specialIssues[key].issue);
```
Comment (joofsh) [inline client/app/containers/EstablishClaimPage/EstablishClaimEmail.jsx]: fancy 🙌

File: client/app/containers/EstablishClaimPage/EstablishClaimEmail.jsx [hunk 5]
```diff
@@ -0,0 +1,106 @@
...
+      selectedSpecialIssue[selectedSpecialIssue.length - 1] =
+        `and ${selectedSpecialIssue[selectedSpecialIssue.length - 1]}`;
+    }
+
+    let email = `The BVA Full Grant decision dated ${formatDate(appeal.decision_date)}` +
+      ` for ${appeal.veteran_name}, ID #${appeal.vbms_id}, was sent to the ARC but` +
+      ` cannot be processed here, as it contains ${selectedSpecialIssue.join(', ')}` +
+      ` in your jurisdiction. Please proceed with control and implement this grant.`;
+
+    this.state = {
```
Comment (joofsh) [inline client/app/containers/EstablishClaimPage/EstablishClaimEmail.jsx]: Nice, I think this makes a lot of sense as state we don't store in the top EstablishClaim 👍

File: client/app/containers/EstablishClaimPage/EstablishClaimReview.jsx [hunk 6]
```diff
@@ -53,13 +53,62 @@ const SPECIAL_ISSUE_NODE_MAP = {
...
 export const UNHANDLED_SPECIAL_ISSUES = [
-  'Pension - United States',
-  'VAMC',
-  'DIC - death, or accrued benefits - United States',
-  'Education - GI Bill, dependents educational ' +
-    'assistance, scholarship, transfer of entitlement',
-  'Waiver of Overpayment',
-  'National Cemetery Administration'
+  {
+    emailAddress: 'PMC',
```
Comment (joofsh) [inline client/app/containers/EstablishClaimPage/EstablishClaimReview.jsx]: how does this work? I would `emailAddress` to always be a valid email. If we aren't using this key for this special issues, maybe better to leave it `null` or `[]`?

File: client/app/util/RoutingConstants.js [hunk 7]
```diff
@@ -0,0 +1,206 @@
+const ROUTING_INFORMATION =
```
Comment (joofsh) [inline client/app/util/RoutingConstants.js]: Can we add comments describing what each layer of this is? I have trouble grokking what  the difference is between `COWC`/ `RO01` / `RO84`

File: client/app/containers/EstablishClaimPage/EstablishClaimEmail.jsx [hunk 8]
```diff
@@ -0,0 +1,106 @@
...
+      map((key) => specialIssues[key].issue);
+
+    // Add an and if there are multiple issues so that the last element
+    // in the list has an and before it.
+    if (selectedSpecialIssue.length > 1) {
+      selectedSpecialIssue[selectedSpecialIssue.length - 1] =
+        `and ${selectedSpecialIssue[selectedSpecialIssue.length - 1]}`;
+    }
+
+    let email = `The BVA Full Grant decision dated ${formatDate(appeal.decision_date)}` +
```
Comment (joofsh) [inline client/app/containers/EstablishClaimPage/EstablishClaimEmail.jsx]: you should be able to use the ES6 multi-line just via backticks http://stackoverflow.com/a/6247331

I think you can just remove the ending backtick and plus, but double check (I could be wrong)

File: client/app/containers/EstablishClaimPage/EstablishClaimReview.jsx (hunk 6 above)
Comment (sharonwarner) [inline client/app/containers/EstablishClaimPage/EstablishClaimReview.jsx]: Yeah, I'm going to fix that in my next PR. There's a separate page that pops up for unhandled special issues that don't have emails.

File: app/models/task.rb [hunk 9]
```diff
@@ -185,6 +188,10 @@ def save_completion_status(status:)
     )
   end
 
+  def no_review_completion_status(status:)
+    status == self.class.completion_status_code(:special_issue_emailed)
```
Comment (joofsh) [inline app/models/task.rb]: hmm, doesn't the guard trigger prior to the state transitoin? If we're currently in the `started` state and trying to transition directly to `completed`, won't this check be false?  Or does the transition & after block execute and then the guard?

**Pull request merged**